src/main/drivers/serial_uart_hal.c

```diff
@@ -64,6 +64,34 @@ static void usartConfigurePinInversion(uartPort_t *uartPort) {
     }
 }
 
+static uartDevice_t *uartFindDevice(uartPort_t *uartPort)
+{
+    for (uint32_t i = 0; i < UARTDEV_COUNT_MAX; i++) {
+        uartDevice_t *candidate = uartDevmap[i];
+
+        if (&candidate->port == uartPort) {
+            return candidate;
+        }
+    }
+    return NULL;
+}
+
+static void uartConfigurePinSwap(uartPort_t *uartPort)
+{
+#if (defined(STM32F1) || defined(STM32F4))
```

Comment:
What about F3?

Reply:
F3 is newer than F1/F4 and has a newer UART peripheral implementation in the MCU.  F3 has pinswap and inversion, F1/F4 do not.  This, and the FPU, is what made the F3 a good choice for flight controllers back in 2015 or so.